### PR TITLE
upgrade flashinfer v0.2.0.post2

### PR DESCRIPTION
**Pull request by zhyncs**
## Motivation

- The flashinfer package name has been changed to flashinfer_python. cc @yzh119 
- Resolve the eagle-related issue. cc @pankajroark
- Temporarily disable the fp8 kv cache test (will be fixed in a follow-up) to facilitate the upgrade.

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).

### CODE REVIEW

File: python/sglang/srt/layers/attention/flashinfer_backend.py [hunk 1]
```diff
@@ -149,6 +149,7 @@ def __init__(
                     BatchPrefillWithPagedKVCacheWrapper(
                         self.workspace_buffer,
                         "NHD",
+                        backend="fa2",
```
Comment (zhyncs) [inline python/sglang/srt/layers/attention/flashinfer_backend.py:152]: fa3 has issues with CUDA Graph @yzh119 cc @merrymercy @Ying1123

File: python/sglang/srt/layers/attention/flashinfer_backend.py [hunk 2]
```diff
@@ -313,7 +314,7 @@ def init_forward_metadata_capture_cuda_graph(
                         paged_kv_indices_buf=self.cuda_graph_kv_indices[i],
                         paged_kv_last_page_len_buf=self.kv_last_page_len[:bs],
                         custom_mask_buf=self.cuda_graph_custom_mask,
-                        qk_indptr_buf=self.cuda_graph_qk_indptr[i][: bs + 1],
+                        mask_indptr_buf=self.cuda_graph_qk_indptr[i][: bs + 1],
```
Comment (zhyncs) [inline python/sglang/srt/layers/attention/flashinfer_backend.py:317]: This is a breaking change from flashinfer @yzh119: the parameter name has been changed.

File: python/sglang/srt/layers/attention/flashinfer_backend.py [hunk 3]
```diff
@@ -1155,41 +1156,24 @@ def fast_decode_plan(
...
-            num_kv_heads,
-            head_dim,
-            page_size,
-            PosEncodingMode[pos_encoding_mode].value,
-            logits_soft_cap,
-            empty_q_data,
-            empty_kv_cache,
-        )
+    stream = torch.cuda.current_stream()
+    self._cached_module.plan(
```
Comment (zhyncs) [inline python/sglang/srt/layers/attention/flashinfer_backend.py:1160]: `_wrapper` has been removed in the latest flashinfer @yzh119 @Ying1123

**Comment by zhyncs** [comment]
BTW In my opinion, the offline FP8 KV cache should be removed and use the online FP8 KV cache, similar to TurboMind. cc @yzh119 @ispobock

**Comment by zhyncs** [comment]
All CIs have passed.

File: python/sglang/srt/layers/attention/flashinfer_backend.py (hunk 1 above)
Comment (momochen) [reply python/sglang/srt/layers/attention/flashinfer_backend.py:152]: Do we have an open issue tracking this?
